### DISCUSSION TITLE
S3 - Upload documents - Field Setting wrong error

### DISCUSSION THREAD

**Discussion by applicationuser** [Budibase/budibase#8677] [Help]
Hello,

Need help with S3 File Upload component.  After placing component in Form field group getting field setting wrong datatype for this component. 

We want to use S3 file component to upload document/pictures to AWS S3 bucket.

![image](https://user-images.githubusercontent.com/42922833/201546684-41cfdb78-3d41-42af-b029-ac26302cdb41.png)

Budibase cloud 
App version : 2.1.25
Table column data type defined as Attachment 

S3 Data source defined without external endpoint 

S3 File component settings - Field setting drop down provided columns of type attachment for selection

![image](https://user-images.githubusercontent.com/42922833/201566161-3bd50e06-6540-4f6b-8998-14008aa2be12.png)

**Comment by melohagan** [comment]
Hey @applicationuser 

The **S3 File Upload** component should have a unique name that does not match any fields in your form schema. If you want to upload files to an attachment field in your table, then you'll need to use the [Attachment](https://docs.budibase.com/docs/attachment) component instead.

**Reply by applicationuser** [reply]
Thank you @melohagan 

S3 Component field name is unique in the table and also in design form

still have the same error , please help

![image](https://user-images.githubusercontent.com/42922833/203086364-0be016a7-4d1d-4549-a228-8ddfc12a5330.png)

**Reply by melohagan** [reply]
Sorry @applicationuser I mean that the field name must be unique. So change `PICTURES` to something that isn't a field in your table.

**Reply by applicationuser** [reply]
Ok @melohagan , got it. 

what do you suggest how we map that uploaded file info to budibase table row ? which can have multiple upload files per row

https://github.com/Budibase/budibase/issues/5454#issuecomment-1102481266   As per this  comment created text field to capture s3 file path.   Created  additional table to hold each row with uploaded file path (multiple pictures can be uploaded)

How can we display to user those uploaded pictures ?  Picture component does not display from S3 url. 

is there any other approach do you suggest ? 

this is what are trying to achieve

- provide user capability to upload multiple pictures
- pictures uploaded to AWS S3 bucket
- capability for user to view pictures
- capability for user to delete pictures

**Reply by melohagan** [reply]
Is there any reason you couldn't use the built-in MinIO instance to store the attachment files? It has better integration within Budibase, including the image component.

**Reply by applicationuser** [reply]
@melohagan  we use budibase cloud

**Comment by melohagan** [comment]
Does this help at all @applicationuser ? https://docs.budibase.com/docs/image#displaying-images-from-an-s3-bucket

**Reply by applicationuser** [reply]
@melohagan 
Yes we can use that by making S3 bucket public access but that might require internal discussion if we want that route,

what if we use standard Attachment component and store uploaded files in budibase cloud.  do we have access to export them out of budibase if required ?
